### PR TITLE
[REEF-425] Create Org.Apache.REEF.IMRU

### PR DESCRIPTION
**Pull request by markusweimer**
This change creates the new project `Org.Apache.REEF.IMRU` to house the new IMRU code to be added. The project creates a NuGet as well.

JIRA:
  [REEF-425](https://issues.apache.org/jira/browse/REEF-425)

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
```
Comment (yingdachen) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: should follow other project files' format using build.props. for example, refer to, say, reef.driver for csproj example

Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: GRML. I was sure to have removed all of this, but VS has added it back in. Let me fix it in the next commit.

Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: Done.

File: lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    <Reference Include="System.Data.DataSetExtensions" />
+    <Reference Include="Microsoft.CSharp" />
+    <Reference Include="System.Data" />
+    <Reference Include="System.Xml" />
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="Properties\AssemblyInfo.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\Org.Apache.REEF.Common\Org.Apache.REEF.Common.csproj">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: use $(SolutionDir) replace ..

File: lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      <Project>{545a0582-4105-44ce-b99c-b1379514a630}</Project>
+      <Name>Org.Apache.REEF.Common</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\Org.Apache.REEF.Driver\Org.Apache.REEF.Driver.csproj">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: use $(SolutionDir) replace ..

File: lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      <Project>{a6baa2a7-f52f-4329-884e-1bcf711d6805}</Project>
+      <Name>Org.Apache.REEF.Driver</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\Org.Apache.REEF.Evaluator\Org.Apache.REEF.Evaluator.csproj">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: same as above

File: lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      <Project>{1b983182-9c30-464c-948d-f87eb93a8240}</Project>
+      <Name>Org.Apache.REEF.Evaluator</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\Org.Apache.REEF.Network\Org.Apache.REEF.Network.csproj">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: same as above

File: lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      <Project>{883ce800-6a6a-4e0a-b7fe-c054f4f2c1dc}</Project>
+      <Name>Org.Apache.REEF.Network</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\Org.Apache.REEF.Tang\Org.Apache.REEF.Tang.csproj">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: same as above

File: lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      <Project>{97dbb573-3994-417a-9f69-ffa25f00d2a6}</Project>
+      <Name>Org.Apache.REEF.Tang</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\Org.Apache.REEF.Utilities\Org.Apache.REEF.Utilities.csproj">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: same as above

File: lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+      <Project>{79e7f89a-1dfb-45e1-8d43-d71a954aeb98}</Project>
+      <Name>Org.Apache.REEF.Utilities</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\Org.Apache.REEF.Wake\Org.Apache.REEF.Wake.csproj">
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IMRU/Org.Apache.REEF.IMRU.csproj]: same as above

**Comment by markusweimer** [comment]
@jwang98052 I addressed your comments. Please have another look.

**Comment by jwang98052** [comment]
NuGet is not created. I will fix it with the merge.

**Pull request closed without merging**